syft/pkg/cataloger/homebrew/parse_homebrew.go

```diff
@@ -0,0 +1,111 @@
+package homebrew
+
+import (
+	"bufio"
+	"context"
+	"path/filepath"
+	"strings"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/internal/unknown"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/generic"
+)
+
+type parsedHomebrewData struct {
+	Name     string
+	Version  string
+	Desc     string
+	Homepage string
+}
+
+func parseHomebrewPackage(_ context.Context, resolver file.Resolver, _ *generic.Environment, reader file.LocationReadCloser) ([]pkg.Package, []artifact.Relationship, error) {
+	var pkgs []pkg.Package
+
+	pathParts := strings.Split(reader.Location.RealPath, string(filepath.Separator))
+	var pkgName, pkgVersion string
+	for i, part := range pathParts {
+		if part == "Cellar" && i+2 < len(pathParts) {
+			pkgName = pathParts[i+1]
+			pkgVersion = pathParts[i+2]
+			break
+		}
+	}
+
+	if pkgName == "" || pkgVersion == "" {
+		return nil, nil, nil
+	}
+
+	pd, err := parseFormulaFile(reader)
+	if err != nil {
+		log.WithFields("package", pkgName).Warn("failed to parse formula")
```

Comment:
Are there any cases where there are multiple `*.rb` files in the directory and we only want to parse a specific one / not return errors for the others? maybe this should return early for glob matches other than `<pkgName>.rb` ?

Reply:
not that i'm aware of (and from a search of what I have installed)